rotkehlchen/globaldb/handler.py

```diff
@@ -254,69 +256,58 @@ def add_setting_value(name: str, value: Any) -> None:
             )
 
     @staticmethod
-    def add_asset(
-            asset_id: str,
-            asset_type: AssetType,
-            data: Union[EvmToken, dict[str, Any]],
-    ) -> None:
+    def add_asset(asset: AssetWithNameAndType) -> None:
         """
-        Add an asset in the DB. Either an ethereum token or a user asset.
-
-        If it's a user asset the data should be typed. As given in by marshmallow.
+        Add an asset in the DB.
 
         May raise InputError in case of error, meaning asset exists or some constraint hit"""
-        if asset_type == AssetType.EVM_TOKEN:
-            token = cast(EvmToken, data)
-            forked_asset = token.forked
-            name = token.name
-            symbol = token.symbol
-            started = token.started
-            swapped_for = token.swapped_for.identifier if token.swapped_for else None
-            coingecko = token.coingecko
-            cryptocompare = token.cryptocompare
-        else:
-            token = None
-            data = cast(dict[str, Any], data)
-            forked_asset = data.get('forked', None)
-            # The data should already be typed (as given in by marshmallow)
-            name = data.get('name', None)
-            symbol = data.get('symbol', None)
-            started = data.get('started', None)
-            swapped_for_asset = data.get('swapped_for', None)
-            swapped_for = swapped_for_asset.identifier if swapped_for_asset else None
-            coingecko = data.get('coingecko', None)
-            cryptocompare = data.get('cryptocompare', '')
-
-        forked = forked_asset.identifier if forked_asset is not None else None
         try:
             with GlobalDBHandler().conn.write_ctx() as write_cursor:
                 write_cursor.execute(
                     'INSERT INTO assets(identifier, name, type) '
                     'VALUES(?, ?, ?);',
                     (
-                        asset_id,
-                        name,
-                        asset_type.serialize_for_db(),
+                        asset.identifier,
+                        asset.name,
+                        asset.asset_type.serialize_for_db(),
                     ),
                 )
-                if token is not None:
-                    GlobalDBHandler.add_evm_token_data(write_cursor, token)
+                if asset.asset_type == AssetType.CUSTOM_ASSET:
+                    write_cursor.execute(
+                        'INSERT INTO custom_assets(identifier, type, notes) VALUES(?, ?, ?)',
+                        cast(CustomAsset, asset).serialize_for_db(),
+                    )
+                    return
+
+                asset = cast(AssetWithOracles, asset)
+                forked, started, swapped_for = None, None, None
+                if asset.is_crypto():
+                    if asset.is_evm_token():
```

Comment:
I made it like this since I remember Lefteris mentioning in some PR that if a function's / variable's name clearly states that it is a boolean, then we should use it without `is True` / `is False`.